pandas/_libs/groupby_helper.pxi.in

```diff
@@ -76,7 +76,11 @@ def group_last_{{name}}({{c_type}}[:, :] out,
                 val = values[i, j]
 
                 # not nan
-                if val == val and val != {{nan_val}}:
+                if (
+                {{if not name.startswith("int")}}
```

Comment:
This is a little hacky - IIRC in other modules we template of the NA value. Is that not an option instead?

Reply:
That's not the warning. The issue is that (val == val) is always true for ints, and the compiler flags it as a possible mistake (since why would you?). this omits this part of the check for `ints`.

Reply:
we don’t do this anywhere else
so rather have this consistently done instead of we decide (separate pr)

Reply:
don't understand.

Reply:
Let's leave this out of this PR for now. To Jeff's point we should be consistent in how we clean up this warning which will require touching quite a few areas. Doing this properly will require proper templating and/or use of fused types

Reply:
no prob. updated.

Reply:
Just out of curioisity, how would fused types help and/or what is improper with this template? it's important, because interval.pyx has similar problem (warnings due to) with fused types covering both signed and unsigned types, and code that does comparisons. If you don't check by name, what would be a cleaner way?

Reply:
I am not discarding the approach just yet but whatever we do should be consistent. If you look in this module there are multiple ways of handling this (see line 391 for an alternate approach) and other modules may have their own handling

So best for a separate comprehensive PR to address and standardize

Reply:
typo? line 391 doesn't seem relevant.

Reply:
https://github.com/pandas-dev/pandas/blob/355e322b5ed4a31c9e1cab9652510e85a54f23a3/pandas/_libs/groupby_helper.pxi.in#L391

Reply:
ah. different commit. I removed these and similar change.

Reply:
You know what? This is really irritating.

My changes elicit a "we don’t do this anywhere else" response from jreback, and "whatever we do should be consistent" from you, when in the meanwhile https://github.com/pandas-dev/pandas/blob/355e322b5ed4a31c9e1cab9652510e85a54f23a3/pandas/_libs/groupby_helper.pxi.in#L56-L60

and https://github.com/pandas-dev/pandas/blob/355e322b5ed4a31c9e1cab9652510e85a54f23a3/pandas/_libs/groupby_helper.pxi.in#L113-L117

and 
https://github.com/pandas-dev/pandas/blob/355e322b5ed4a31c9e1cab9652510e85a54f23a3/pandas/_libs/groupby_helper.pxi.in#L212-L216

and multiple others. 

Why reject fixes for an actual problem, in the name of a code style that may or may not someday come to be? It'll probably be this very same thing, by the look of it.

Reply:
Right those are inconsistent and none match what you have in this PR. I understand the frustration (I've introduced some of that inconsistency) but we are just trying to improve the code base

In any case certainly would be welcome the fix but separate PR as it needs a more comprehensive look on the commit and review side of things

Reply:
argh. moving on.